### PR TITLE
Add job runner for Chronos

### PR DESCRIPTION
**Pull request by theosotr**
Chronos (https://mesos.github.io/chronos/) is a framework for the Apache Mesos (http://mesos.apache.org/); a software which manages computer clusters. Specifically,
Chronos is a batch processing scheduler that runs on top of Mesos and it is used for job
orchestration.

### CODE REVIEW

File: config/job_conf.xml.sample_advanced [hunk 1]
```diff
@@ -703,6 +722,17 @@
...
             -->
         </destination>
+        <destination id="chronos_dest" runner="chronos">
+            <param id="docker_enabled">true</param>
+            <param id="docker_memory">512</param>
+            <param id="docker_cpu">2</param>
+            <param id="volumes">/directory/</param>
+            <!-- Directory which is mounted to the container and is parent of
+                 the  `job_working_directory`, `file_path`, `new_file_path`
+                 directories -->
```
Comment (jmchilton) [inline config/job_conf.xml.sample_advanced]: Can you update this comment to mention reference data as well - everything mentioned in ``.loc`` files (if they are used by these jobs) need to be included as well.

**Comment by jmchilton** [comment]
This is utterly fantastic - while I think that we could open up the deployment opportunities by integrating this with Pulsar instead since we could map everything into that ``volumes`` directory as needed - I really do appreciate you very clearly lay out the requirements to use this job runner and avoided hacks to do file staging (which Galaxy makes difficult).

Once we get that client library in starforge - I'll be a +1 on this. I don't think we have how to do that documented anywhere though - so I can try to find some time to do it.

**Comment by jmchilton** [comment]
Thanks for the starforge PR @bgruening - the wheel is now available at https://wheels.galaxyproject.org/packages/chronos-python-0.38.0.tar.gz#md5=6d64e6d101c80c50c0d7add57e23c9bd.

File: lib/galaxy/dependencies/pinned-requirements.txt [hunk 2]
```diff
@@ -69,3 +69,6 @@ ecdsa==0.13
 
 # Flexible BAM index naming
 pysam==0.8.4+gx5
+
+# Chronos client
+chronos-python
```
Comment (jmchilton) [inline lib/galaxy/dependencies/pinned-requirements.txt]: Will you pin the version (use the line ``chronos-python==0.38.0`` instead) please?

**Comment by bgruening** [comment]
@jmchilton was this a +1 from you?
For me this looks great, it's documented and isolated enough to get it in.

**Comment by nsoranzo** [comment]
`chronos-python` has a dependency on `httplib2>=0.9`, which is not available in https://wheels.galaxyproject.org/ and is not listed in our requirements files.

Also, these 2 requirements can maybe go under `lib/galaxy/dependencies/conditional-requirements.txt` instead of `pinned-requirements.txt`, with the appropriate code added to `lib/galaxy/dependencies/__init__.py`, what do you think?

**Comment by jmchilton** [comment]
@nsoranzo - yup - these need to be done. I've created an issue for this https://github.com/galaxyproject/galaxy/issues/4085.

**Comment by nsoranzo** [comment]
Thanks @jmchilton!

**Comment by jmchilton** [comment]
@nsoranzo Out of curiosity how did you notice? Is there some automated check we could quickly implement to ask if Galaxy's dependencies are all available as ~~wheels~~ Galaxy-hosted wheels?

**Comment by nsoranzo** [comment]
I noticed my dev Galaxy downloaded 2 new wheels after a `git pull`, I remembered this PR but not about `httplib2`, went to check on https://wheels.galaxyproject.org/ and didn't find it, so manual :eagle: :eye: process ;)

**Pull request merged**